vega_sim/api/data.py

```diff
@@ -397,6 +397,144 @@ class TeamRefereeHistory:
+
+
+@dataclass(frozen=True)
+class StopOrderEvent:
+    submission: OrderSubmission
+    stop_order: StopOrder
+
+
+def _pegged_order_from_proto(pegged_order, decimal_spec: DecimalSpec) -> PeggedOrder:
+    return PeggedOrder(
+        reference=pegged_order.reference,
+        offset=num_from_padded_int(pegged_order.offset, decimal_spec.price_decimals),
+    )
+
+
+def _iceberg_opts_from_proto(iceberg_opts, decimal_spec: DecimalSpec) -> IcebergOpts:
+    return IcebergOpts(
+        peak_size=num_from_padded_int(
+            iceberg_opts.peak_size, decimal_spec.position_decimals
+        ),
+        minimum_visible_size=num_from_padded_int(
+            iceberg_opts.minimum_visible_size, decimal_spec.position_decimals
+        ),
+    )
+
+
+def _order_submission_from_proto(
+    order_submission: vega_protos.commands.OrderSubmission,
+    decimal_spec: DecimalSpec,
+) -> OrderSubmission:
+    return OrderSubmission(
+        market_id=order_submission.market_id,
+        price=num_from_padded_int(order_submission.price, decimal_spec.price_decimals),
+        size=num_from_padded_int(order_submission.size, decimal_spec.position_decimals),
+        side=order_submission.side,
+        time_in_force=order_submission.time_in_force,
+        expires_at=datetime.datetime.fromtimestamp(order_submission.expires_at / 1e9)
+        if order_submission.expires_at != 0
+        else None,
+        type=order_submission.type,
+        reference=order_submission.reference,
+        pegged_order=_pegged_order_from_proto(
+            order_submission.pegged_order, decimal_spec
+        ),
+        post_only=order_submission.post_only,
+        reduce_only=order_submission.reduce_only,
+        iceberg_opts=_iceberg_opts_from_proto(
+            order_submission.iceberg_opts, decimal_spec
+        ),
+    )
+
+
+def _stop_order_from_proto(
+    stop_order: vega_protos.vega.StopOrder, decimal_spec: DecimalSpec
+) -> StopOrder:
+    return StopOrder(
+        id=stop_order.id,
+        oco_link_id=stop_order.oco_link_id,
+        expires_at=datetime.datetime.fromtimestamp(stop_order.expires_at / 1e9)
+        if stop_order.expires_at != 0
+        else None,
+        expiry_strategy=stop_order.expiry_strategy,
+        trigger_direction=stop_order.trigger_direction,
+        status=stop_order.status,
+        created_at=datetime.datetime.fromtimestamp(stop_order.created_at / 1e9)
+        if stop_order.created_at != 0
+        else None,
+        updated_at=datetime.datetime.fromtimestamp(stop_order.updated_at / 1e9)
+        if stop_order.updated_at != 0
+        else None,
+        order_id=stop_order.order_id,
+        party_id=stop_order.party_id,
+        market_id=stop_order.market_id,
+        rejection_reason=stop_order.rejection_reason,
+        price=num_from_padded_int(stop_order.price, decimal_spec.price_decimals)
+        if stop_order.price is not None
+        else None,
+        trailing_percent_offset=stop_order.trailing_percent_offset
```

Comment:
Does this come in as a float or does it need converting from a string?

Reply:
If the docs are correct, it should be decimal value:

> This should be expressed as a decimal value between 0 and 1, e.g. 0.01 for 1%

Which I think is the right format for us? Unless you think it is misleading having the word percent but using a float?